homeassistant/components/switch/edimax.py

```diff
@@ -0,0 +1,75 @@
+"""
+homeassistant.components.switch.edimax
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Support for Edimax switches.
+"""
+import logging
+
+from homeassistant.components.switch import SwitchDevice
+from homeassistant.const import CONF_HOST, CONF_USERNAME, CONF_PASSWORD
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+    """ Find and return Edimax Smart Plugs. """
+    try:
+        # pylint: disable=no-name-in-module, import-error
+        from pyedimax.smartplug import SmartPlug
+    except ImportError:
+        logging.getLogger(__name__).exception((
+            "Failed to import pyedimax. "))
+
+        return
+
+    host = config.get(CONF_HOST)
+    auth = (config.get(CONF_USERNAME, 'admin'),
+            config.get(CONF_PASSWORD, '1234'))
+
+    if not host:
+        logging.getLogger(__name__).error(
+            'Missing config variable %s', CONF_HOST)
+        return False
+
+    add_devices_callback([SmartPlugSwitch(SmartPlug(host, auth))])
+
+
+class SmartPlugSwitch(SwitchDevice):
+    """ Represents an Edimax Smart Plug switch within Home Assistant. """
+    def __init__(self, smartplug):
+        self.smartplug = smartplug
+
+    @property
+    def name(self):
+        """ Returns the name of the Smart Plug, if any. """
+        #TODO: dynamically get name from device using requests
+        return 'Edimax Smart Plug'
```

Comment:
Todo comments are not accepted into the repository. Please change this to future. It may be nice to at least have a config option to set the name of the device if the name is not pulled from the device.